### ISSUE TITLE
Use training steps instead of epochs in val/train logs

### ISSUE THREAD

**Issue by nickiliak** [nickiliak/Floorplan-Diffusion-Models#3]
## Description

Validation and training logging intervals are configured in epochs (`check_val_every_n_epoch`) while every other interval in the codebase is step-based. This is inconsistent with the HouseDiffusion reference implementation, which is entirely step-driven, and with our own Lightning module's LR scheduler (already set to `"interval": "step"`).

## Steps to Reproduce

1. Open `scripts/train.py` and inspect line 198 — the `Trainer` is constructed with `check_val_every_n_epoch=cfg.check_val_every_n_epoch`.
2. Open any config file (`configs/default.yaml`, `configs/test.yaml`, `configs/overfit.yaml`) — all define `check_val_every_n_epoch` (values: 40, 5, 40).
3. Compare with other intervals in the same configs: `max_steps`, `save_interval`, `log_interval`, `warmup_steps`, `lr_decay_steps` — all step-based.
4. Compare with the HouseDiffusion reference (`external/house_diffusion/house_diffusion/train_util.py`) — the training loop is entirely step-driven (`self.step % self.log_interval`, `self.step % self.save_interval`), with no epoch concept.
5. Inspect `src/floorplan_diffusion/model/lightning_module.py` line 154 — the LR scheduler already uses `"interval": "step"`.

## Expected Behavior

All logging and validation intervals should be expressed in training steps, consistent with:
- The rest of the codebase's interval parameters
- The HouseDiffusion reference implementation
- The Lightning module's own LR scheduler interval

The `Trainer` should use `val_check_interval` (step-based) instead of `check_val_every_n_epoch`.

## Actual Behavior

`scripts/train.py:198` passes `check_val_every_n_epoch` to the Lightning `Trainer`, making validation frequency depend on epoch boundaries. This is the sole epoch-based interval in the entire codebase:

| Parameter | Unit | Source |
|---|---|---|
| `max_steps` | steps | `scripts/train.py` |
| `save_interval` | steps | configs |
| `log_interval` | steps | configs |
| `warmup_steps` | steps | configs |
| `lr_decay_steps` | steps | configs |
| `check_val_every_n_epoch` | **epochs** | configs / `train.py:198` |

Because epoch length depends on dataset size and batch size, the actual step interval between validations is unpredictable and changes silently when data or batch config changes.

## Relevant Code

- `scripts/train.py:198` — `Trainer(check_val_every_n_epoch=cfg.check_val_every_n_epoch, ...)`
- `configs/default.yaml` — `check_val_every_n_epoch: 40`
- `configs/test.yaml` — `check_val_every_n_epoch: 5`
- `configs/overfit.yaml` — `check_val_every_n_epoch: 40`
- `src/floorplan_diffusion/model/lightning_module.py:154` — LR scheduler uses `"interval": "step"`
- `external/house_diffusion/house_diffusion/train_util.py` — reference impl is fully step-driven

## Suggested Fix

1. In `scripts/train.py`, replace `check_val_every_n_epoch` with `val_check_interval` in the `Trainer` constructor.
2. In all 3 config files, rename the key from `check_val_every_n_epoch` to `val_check_interval` and convert values from epoch counts to equivalent step counts.
3. Verify that the `val_check_interval` values align with `save_interval` and `log_interval` for a consistent logging cadence.

**Comment by nickiliak** [comment]
## 🔍 Investigation: Use training steps instead of epochs in val/train logs

**Type**: `BUG`

### Assessment

| Metric | Value | Reasoning |
|--------|-------|-----------|
| Severity | `MEDIUM` | Validation still runs, but at unpredictable intervals that silently change when batch size or dataset size changes |
| Complexity | `LOW` | 4 files changed (1 train script + 3 configs), all changes are key renames/value substitutions with no logic changes |
| Confidence | `HIGH` | Root cause is a single argument in `scripts/train.py:198`; all affected files are clearly identified and changes are mechanical |

---

### Problem Statement

`scripts/train.py` constructs the Lightning `Trainer` with `check_val_every_n_epoch`, making validation frequency epoch-dependent while every other interval in the codebase (save, log, LR warmup, LR decay, max training) is step-based. Epoch length varies with dataset size and batch size, so the actual step count between validations is unpredictable and changes silently when data or batch config changes.

---

### Root Cause Analysis

WHY: Validation runs at unpredictable step intervals
↓ BECAUSE: `Trainer` uses `check_val_every_n_epoch` (epoch boundary trigger)
  → `scripts/train.py:198` — `check_val_every_n_epoch=train_cfg["check_val_every_n_epoch"]`

↓ ROOT CAUSE: `val_check_interval` (Lightning's step-based alternative) was never used
  → Contrast: `ModelCheckpoint` already uses `every_n_train_steps=train_cfg["save_interval"]` correctly at `scripts/train.py:187`

**Introduced in**: `3e4ca8e` — "chore: split configs into def/test and document hyperparameters"

---

### Implementation Plan

| Step | File | Change |
|------|------|--------|
| 1 | `scripts/train.py:198` | Replace `check_val_every_n_epoch=...` with `val_check_interval=train_cfg["val_check_interval"]` |
| 2 | `configs/resplan_housediff_def.yaml:24` | `check_val_every_n_epoch: 40` → `val_check_interval: 10000` (align with `save_interval`) |
| 3 | `configs/resplan_housediff_stable_fp32.yaml:21` | `check_val_every_n_epoch: 40` → `val_check_interval: 25000` (align with `save_interval`) |
| 4 | `configs/resplan_housediff_test.yaml:25` | `check_val_every_n_epoch: 5` → `val_check_interval: 250` (align with `save_interval`, gives 2 val runs in 500-step smoke test) |

<details>
<summary>📋 Detailed Implementation Steps</summary>

**Step 1 — `scripts/train.py:196-206`**

```python
# Before
trainer = pl.Trainer(
    max_steps=train_cfg["max_steps"],
    check_val_every_n_epoch=train_cfg["check_val_every_n_epoch"],
    ...
)

# After
trainer = pl.Trainer(
    max_steps=train_cfg["max_steps"],
    val_check_interval=train_cfg["val_check_interval"],
    ...
)
```

**Step 2 — `configs/resplan_housediff_def.yaml:24`**
```yaml
# Before
  check_val_every_n_epoch: 40
# After
  val_check_interval: 10000
```

**Step 3 — `configs/resplan_housediff_stable_fp32.yaml:21`**
```yaml
# Before
  check_val_every_n_epoch: 40
# After
  val_check_interval: 25000
```

**Step 4 — `configs/resplan_housediff_test.yaml:25`**
```yaml
# Before
  check_val_every_n_epoch: 5
# After
  val_check_interval: 250
```

</details>

---

### Validation

```bash
uv run task lint
uv run task test
# Manual: uv run task train --config configs/resplan_housediff_test.yaml
# Verify val/loss appears in metrics.csv at steps 250 and 500
```

---

### Next Step

To implement: `/implement-issue 3`

---
*Investigated by Claude • 2026-04-28*

**Comment by nickiliak** [comment]
## ✅ Issue Resolution Report

**PR**: #10 (https://github.com/nickiliak/Floorplan-Diffusion-Models/pull/10)
**Status**: COMPLETE

---

### Summary

Root cause was `scripts/train.py:198` passing `check_val_every_n_epoch` to `pl.Trainer` instead of `val_check_interval`, while every other training interval in the codebase (`save_interval`, `log_interval`, `warmup_steps`, `lr_decay_steps`) is step-based. The fix renames the Trainer kwarg and updates all three YAML configs to use step counts aligned with each config's `save_interval`.

---

### Changes Made

| File | Change |
|------|--------|
| `scripts/train.py` | Replaced `check_val_every_n_epoch` with `val_check_interval` in `pl.Trainer` constructor |
| `configs/resplan_housediff_def.yaml` | `val_check_interval: 10000` (was `check_val_every_n_epoch: 40`) |
| `configs/resplan_housediff_stable_fp32.yaml` | `val_check_interval: 25000` (was `check_val_every_n_epoch: 40`) |
| `configs/resplan_housediff_test.yaml` | `val_check_interval: 250` (was `check_val_every_n_epoch: 5`) |
| `tests/test_step7_validation.py` | 13 new test cases guarding config key contract and Trainer kwarg |

---

### Validation

✅ Lint (0 errors) | ✅ Format | ✅ Tests (35 passed, 3 skipped)

---

### Review & Self-Fix

- **5** review findings identified (1 CRITICAL, 1 HIGH, 3 LOW)
- **4** findings fixed (core rename, missing tests, spurious noqa)
- **1** skipped (LOW: `extract_data.py` has no tests — out of scope)

---

### Unaddressed Items

| Finding | Severity | Reason |
|---------|----------|--------|
| `extract_data.py` has no test coverage | LOW | Only cosmetically reformatted in this PR; adding tests is a new feature |

---

### Suggested Follow-up Issues

1. **"Add config schema validation (Pydantic) for training YAML"** (P2) — prevent silent KeyError-at-runtime for config key renames
2. **"Add unit tests for extract_data.py"** (P3) — script has no test coverage

---

*Resolved by Archon workflow `d4a3fac85ccad2dd64557ac1d32008b7`*
